desktop/webpack.config.js

```diff
@@ -0,0 +1,51 @@
+/**
```

Comment:
Pending resolution:

TL;DR the unpackaged app is running without problems from the project root ("desktop" subfolder of Calypso repo), but not resolving correctly when packaged.

#### Details

If the `main` entry in package.json ([L16](https://github.com/Automattic/wp-calypso/pull/49283/files#diff-070ef7659160d7a5d6678b3736e57b45a08caf60f3ceee146e6f7027292de0f7R16)) is changed to `"main": "app/index.js"`, you can run the app with a simple `npx electron .` (from the `calypso/desktop` folder).

However, even though `yarn run build` completes with no errors, running the _built_ (packaged) app is unsuccessful and fails with the error `r is not a function` (see screenshot):

<p align="center">
<img src="https://user-images.githubusercontent.com/8979548/105788629-1cc1b500-5f3e-11eb-86b2-577198f4ee74.png">
</p>

This is because something in the webpack configuration isn't quite right and is not getting resolved correctly. Note: the app is built to `desktop/release/mac`. Also, you can use `ELECTRON_BUILDER_ARGS=-c.mac.target=dir yarn run build` to build only the app without the installers, which saves some time.

Would appreciate some help from someone with more Webpack experience to help troubleshoot.

Reply:
"build:main": "webpack --display errors-only",

WARNING in ../node_modules/formidable/lib/incoming_form.js 1:43-50
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
 @ ../node_modules/formidable/lib/index.js
 @ ../node_modules/superagent/lib/node/index.js
 @ ./lib/crash-tracker/index.js
 @ ./app-handlers/exceptions/index.js
 @ ./env.js
 @ ./app.js
 @ ./index.js

WARNING in ../node_modules/ws/lib/buffer-util.js
Module not found: Error: Can't resolve 'bufferutil' in '/Users/jb/repos/wp-calypso/desktop/node_modules/ws/lib'
 @ ../node_modules/ws/lib/buffer-util.js
 @ ../node_modules/ws/lib/sender.js
 @ ../node_modules/ws/index.js
 @ ./lib/notifications/api/index.js
 @ ./window-handlers/login-status/index.js
 @ ./server/index.js
 @ ./app.js
 @ ./index.js

WARNING in ../node_modules/ws/lib/validation.js
Module not found: Error: Can't resolve 'utf-8-validate' in '/Users/jb/repos/wp-calypso/desktop/node_modules/ws/lib'
 @ ../node_modules/ws/lib/validation.js
 @ ../node_modules/ws/lib/receiver.js
 @ ../node_modules/ws/index.js
 @ ./lib/notifications/api/index.js
 @ ./window-handlers/login-status/index.js
 @ ./server/index.js
 @ ./app.js
 @ ./index.js

Reply:
Well that isn't it.

Reply:
This is causing the initial error, comment out.

https://github.com/Automattic/wp-calypso/pull/49283/files#diff-27be89e385ba877fbcb9958f2dc963940bd1e0487cb982581d703458fa1e6b71R28

Reply:
Comment out all these requires and no more error:

https://github.com/Automattic/wp-calypso/pull/49283/files#diff-ddc3c9494313400e0157b6bbf9c19c29f11102ebc3bdbb28dd8030f830bf1e72R112

Reply:
Further narrowed the errors to these two lines:

	// require( '../window-handlers/login-status' )( mainWindow );
	// require( '../window-handlers/notifications' )( mainWindow );

Reply:
in `window-handlers/login-status/index.js`

This is what is causing the issue.

const WPNotificationsAPI = require( '../../lib/notifications/api' );

Reply:
I think you will have to just keep digging till you find the root cause. I didn't see anything wrong with the webpack config.

Reply:
desktop/app/window-handlers/login-status/index.js comment out:
// const WPNotificationsAPI = require( '../../lib/notifications/api' );
// await WPNotificationsAPI.connect();

desktop/app/server/index.js comment out:
	// require( '../window-handlers/notifications' )( mainWindow );

desktop/app/env.js comment out:
// require( './app-handlers/exceptions' )();